src/Xamarin.Android.Build.Utilities/Sdks/AndroidSdkWindows.cs

```diff
@@ -24,27 +24,36 @@ class AndroidSdkWindows : AndroidSdkBase
 		public override string NdkHostPlatform64Bit { get { return "windows-x86_64"; } }
 		public override string Javac { get; protected set; } = "javac.exe";
 
+		string GetMDRegistryKey ()
+		{
+			var regKey = Environment.GetEnvironmentVariable ("XAMARIN_ANDROID_REGKEY");
+			return string.IsNullOrWhiteSpace (regKey) ? MDREG_KEY : regKey;
+		}
+
 		public override string PreferedAndroidSdkPath {
 			get {
 				var wow = RegistryEx.Wow64.Key32;
-				if (CheckRegistryKeyForExecutable (RegistryEx.CurrentUser, MDREG_KEY, MDREG_ANDROID_SDK, wow, "platform-tools", Adb))
-					return RegistryEx.GetValueString (RegistryEx.CurrentUser, MDREG_KEY, MDREG_ANDROID_SDK, wow);
+				var regKey = GetMDRegistryKey ();
+				if (CheckRegistryKeyForExecutable (RegistryEx.CurrentUser, regKey, MDREG_ANDROID_SDK, wow, "platform-tools", Adb))
```

Comment:
> If the key is specified, it must exist, otherwise, it's an error (from the IDE most likely)

This likewise sounds good. Which means we need some form of error reporting so that the IDE/IDE maintainers (i.e. *you*) are notified when such a mistake is made.